src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.CurlResponseMessage.cs

```diff
@@ -393,24 +393,25 @@ internal void SignalComplete(Exception error = null, bool forceCancel = false)
 
                     // If there's a pending read request, complete it, either with 0 bytes for success
                     // or with the exception/CancellationToken for failure.
-                    if (_pendingReadRequest != null)
+                    ReadState pendingRead = _pendingReadRequest;
+                    if (pendingRead != null)
                     {
                         if (_completed == s_completionSentinel)
                         {
                             EventSourceTrace("Completing pending read with 0 bytes");
-                            _pendingReadRequest.TrySetResult(0);
+                            pendingRead.TrySetResult(0);
                         }
                         else
                         {
                             EventSourceTrace("Failing pending read task with error: {0}", _completed);
                             OperationCanceledException oce = _completed as OperationCanceledException;
                             if (oce != null)
                             {
-                                _pendingReadRequest.TrySetCanceled(oce.CancellationToken);
+                                pendingRead.TrySetCanceled(oce.CancellationToken);
                             }
                             else
                             {
-                                _pendingReadRequest.TrySetException(MapToReadWriteIOException(_completed, isRead: true));
+                                pendingRead.TrySetException(MapToReadWriteIOException(_completed, isRead: true));
                             }
                         }
 
```

Comment:
@stephentoub sounds good - I do not know this code too much, I was wondering if we would override some important data by assigning null